x-pack/plugins/observability/public/components/shared/exploratory_view/header/embed_action.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiPopover, EuiCodeBlock, EuiPopoverTitle } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import { TypedLensByValueInput } from '@kbn/lens-plugin/public';
+import { useSeriesStorage } from '../hooks/use_series_storage';
+
+export function EmbedAction({
+  lensAttributes,
+}: {
+  lensAttributes: TypedLensByValueInput['attributes'] | null;
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const { reportType, allSeries } = useSeriesStorage();
+
+  const button = (
+    <EuiButtonEmpty
+      size="s"
+      isDisabled={lensAttributes === null}
+      onClick={() => {
+        setIsOpen(!isOpen);
+      }}
+    >
+      {EMBED_LABEL}
+    </EuiButtonEmpty>
+  );
+
+  return (
+    <EuiPopover button={button} isOpen={isOpen} closePopover={() => setIsOpen(false)}>
+      <EuiPopoverTitle>{EMBED_TITLE_LABEL}</EuiPopoverTitle>
+      <EuiCodeBlock
+        language="jsx"
+        fontSize="m"
+        paddingSize="m"
+        isCopyable={true}
+        style={{ width: 500 }}
+      >
+        {`const { observability } = useKibana<>().services;
```

Comment:
What's the use case for this code? I was thinking it might be good to add a `return ()` here, or set the JSX call to a symbol like `const component =`?

```typescript
{`const { observability } = useKibana().services;

const { ExploratoryViewEmbeddable } = observability;

return (
  <ExploratoryViewEmbeddable
    customHeight={'300px'}
    reportType="${reportType}"
    attributes={${JSON.stringify(allSeries, null, 4)}}
  />
);`}
```

Reply:
main use case is like when you added page views in user experience app and also like i have this PR 

https://github.com/elastic/kibana/pull/135296

so essentially i can create the viz and copy the code easily.